### PR TITLE
Remove .ENV files, update readme

### PR DESCRIPTION
**Pull request by alxjrvs**
This PR 
 - Updates the Readme with proper setup and deploy instructions (albeit fairly barebones) 
 - Removes unused `.env` files and `dotenv` gem (embracing `master.key` makes these superfluous; keeping them around might encourage someone to use them)
 - convinces test environment to make use of rails secrets
 - alters `config/database.yml` to define test values directly

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -43,18 +44,14 @@ jobs:
...
-        run: bundle exec rspec
+        run: |
+          bin/setup
+          bin/rails assets:precompile
+          bundle exec rspec
 
       - name: Run JS Tests
         if: steps.check_files.outputs.files_exists == 'true'
+        env:
+          RAILS_MASTER_KEY: ${{ secrets.RAILS_MASTER_KEY }}
```
Comment (efertsch) [inline .github/workflows/test.yml:56]: 👍

File: README.md [hunk 2]
```diff
@@ -8,17 +8,18 @@ a 🥇 GILD production
...
 
+1. Acquire the `config/master.key` file from a trusted associate.
 1. `bin/setup`
 1. `bin/dev`
 
-Note: `master.key` is not required for this app.
+## Secrets
 
-## Heroku
+GEAR embraces the [Rails 7 Credentials System](https://edgeguides.rubyonrails.org/security.html#custom-credentials). Once you have the `config/master.key`,
```
Comment (efertsch) [inline README.md:17]: Is this last sentence a typo or is is supposed to be a lead into the next section?

File: README.md [hunk 3]
```diff
@@ -8,17 +8,18 @@ a 🥇 GILD production
...
 
-### Running Heroku locally
+## Hosting
 
-Use Procfile.dev and .env.development when running `heroku local`. For example:
+GEAR is automatically deployed as a result of successful checks running on `MAIN` following a merge.
```
Comment (efertsch) [inline README.md:21]: So just to be clear there is no secondary (staging) instance running the app?

File: README.md (hunk 2 above)
Comment (alxjrvs) [reply README.md:17]: Woops, this is me stopping mid sentence and not noticing.

File: README.md (hunk 3 above)
Comment (alxjrvs) [reply README.md:21]: Not currently, no

File: README.md (hunk 2 above)
Comment (alxjrvs) [reply README.md]: [ccf397a](https://github.com/TheGnarCo/GEAR/pull/134/commits/ccf397a73dbff88aadc73fc3ff304f891e6af494)

**Review by efertsch — APPROVED** [review]

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -43,18 +44,14 @@ jobs:
...
-      - name: Generate Assets
-        shell: bash
-        run: bin/rails assets:precompile
-
       - name: Run Ruby Tests
-        shell: bash
-        run: bundle exec rspec
+        run: |
+          bin/setup
+          bin/rails assets:precompile
```
Comment (taylorkearns) [inline .github/workflows/test.yml:50]: This command gets called in `bin/setup`
https://github.com/TheGnarCo/GEAR/blob/main/bin/setup#L32

File: README.md [hunk 5]
```diff
@@ -8,17 +8,22 @@ a 🥇 GILD production
...
+## Secrets
 
-## Heroku
+GEAR embraces the [Rails 7 Credentials System](https://edgeguides.rubyonrails.org/security.html#custom-credentials).
 
-### Running Heroku locally
+Once you have the `config/master.key`, you can run `bin/rails credentials:edit` to view and change the rails secrets.
 
-Use Procfile.dev and .env.development when running `heroku local`. For example:
+(As a reminder: This will change the `credentials.enc.yml` file, which is checked into git.)
```
Comment (taylorkearns) [inline README.md:21]: This is tangential, but: how does this work? The master key is passed around between environments, but it contains environment variables?

Comment (alxjrvs) [reply README.md:21]: @taylorkearns the `credentials.enc.yaml` file has the environment variables, and the `master.key` decrypts it for the various environments. We check in the former, we do not check in the latter.

File: .github/workflows/test.yml (hunk 4 above)
Comment (alxjrvs) [reply .github/workflows/test.yml:50]: ooo, good catch

**Pull request merged**